pkg/util/soft_append_error_processor.go

```diff
@@ -0,0 +1,111 @@
+package util
```

Comment:
This would neatly go into pkg/util/validation, I think? Anything to get it out of "util" which I like to read as "pkg/code". 😄

Reply:
Many errors are specific to tsdb, so I don't think it should go inside `validation`. How about `pkg/storage`?